File: src/gameset/pieces/pawn.rs

use super::super::{Piece, PieceData, Board, Color, Position};

const SYMBOL: char = 'p';
const VALUE: i32 = 100;

pub struct Pawn {
    piece_data: PieceData
}

impl Piece for Pawn {
    fn piece(&self) -> &PieceData {
        &self.piece_data
    }

    fn generate(&self, board: &Board) -> Vec<Board> {
        Vec::new()
    }
}

impl Pawn {
    pub fn new(color: Color,
               position: Position
        ) -> Pawn {
        Pawn {
            piece_data: PieceData::new(
                color,
                SYMBOL,
                VALUE,
                position)
        }
    }
}

File: src/main/java/st/netb/chess/lib/piece/Pawn.java
package st.netb.chess.lib.piece;

import st.netb.chess.lib.Board;

import java.awt.*;
import java.util.List;

public class Pawn extends Piece{

	public Pawn(Piece.Kind kind, Piece.Color color, Point position) {
		super(kind, color, position);
	}

	public Pawn(Piece.Color color, Point position) {
		super(Kind.PAWN, color, position);
	}

	@Override
	public List<Point> allPossibleMoves(Board board) {
		return null;
	}
}
